src/commands/functions-secrets-set.ts

```diff
@@ -36,15 +39,50 @@ export const command = new Command("functions:secrets:set <KEY>")
     "--data-file <dataFile>",
     'file path from which to read secret data. Set to "-" to read the secret data from stdin',
   )
+  .option("--format <format>", "format of the secret value. 'string' (default) or 'json'")
   .action(async (unvalidatedKey: string, options: Options) => {
     const projectId = needProjectId(options);
     const projectNumber = await needProjectNumber(options);
     const key = await ensureValidKey(unvalidatedKey, options);
     const secret = await ensureSecret(projectId, key, options);
-    const secretValue = await readSecretValue(
-      `Enter a value for ${key}`,
-      options.dataFile as string | undefined,
-    );
+
+    // Determine format using priority order: explicit flag > file extension > default to string
+    let format = options.format as string | undefined;
+    const dataFile = options.dataFile as string | undefined;
+    if (!format && dataFile && dataFile !== "-") {
+      // Auto-detect from file extension
+      if (dataFile.endsWith(".json")) {
+        format = "json";
+      }
+    }
+
+    // Read secret value from file, stdin, or interactive prompt
+    let secretValue: string;
+    if (dataFile) {
+      // Read from file or stdin
+      const inputSource = dataFile === "-" ? 0 : dataFile;
+      try {
+        secretValue = fs.readFileSync(inputSource, "utf-8");
+      } catch (e: any) {
+        if (e.code === "ENOENT") {
+          throw new FirebaseError(`File not found: ${inputSource}`, { original: e });
+        }
+        throw e;
+      }
+    } else if (!tty.isatty(0)) {
```

Comment:
There's a `nonInteractive` variable passed around in params.ts. Could that be used here instead of a separate check?

Reply:
oh great catch. fixed.